Версия № 6:

ZotovSlava opened the pull request:
1) Добавлена возможность устанавливать время начала и окончания выполнения задач. 2) Добавлена возможность выводить задачи в порядке приоритета по времени. 3) Задачи время которых пересекается не добавляются. 4) Добавлены новые тесты.

ZotovSlava committed:
Версия № 6:
1) Добавлена возможность устанавливать время начала и окончания выполнения задач.
2) Добавлена возможность выводить задачи в порядке приоритета по времени.
3) Задачи время которых пересекается не добавляются.
4) Добавлены новые тесты.

ZotovSlava committed:
Версия № 6.1:
1) Исп. GitTest

litvitnik commented on src/InMemoryTaskManager.java:
```diff
@@ -213,4 +280,24 @@ private void calculateEpicStatus(int epicId) {
         }
     }
 
+    private void calculateEpicsTime(int epicId) {
```
Этот метод должен находиться в классе Epic. Также необходимо внести исправления в этот метод. В техническом задании указано, что продолжительность эпика должна рассчитываться как сумма продолжительностей всех его подзадач. Однако в текущей реализации продолжительность устанавливается только в случае, если у эпика есть лишь одна подзадача, а в случае нескольких подзадач она определяется на основе начального и конечного времени. В общем, ты правильно обновляешь startTime и endTime, но не суммируешь продолжительность всех подзадач для определения duration.

ZotovSlava committed:
Версия № 6.2:
1) Метод calculateEpicsTime перенесен в класс Epic.
2) Изменена логика расчета времени необходимого для выполнения Epic. Теперь это сумма времени выполнения всех его Subtasks.

ZotovSlava committed:
Версия № 6.2:
1) Метод calculateEpicsTime перенесен в класс Epic.
2) Изменена логика расчета времени необходимого для выполнения Epic. Теперь это сумма времени выполнения всех его Subtasks.

ZotovSlava committed:
Версия № 6.3:
1) Изм. названия метода
